### ISSUE TITLE
MPIIGaze Dataset - is scaling factor applied to the normalized gaze?

### ISSUE THREAD

**Issue by 1cookspe** [xucong-zhang/data-preprocessing-gaze#1]
Hello Professor Xucong Zhang,

Thank you very much for providing the data preprocessing gaze - it has helped us immensely with preprocessing our own gaze data. We have trained a neural network inspired by [1] to predict gaze from the user's eyes, and we used the normalized data in the MPIIGaze dataset for training. Specifically, we used the data in the "Normalized" folder from the MPIIGaze dataset obtained from the Perceptual User Interfaces website at [https://perceptualui.org/research/datasets/MPIIGaze/](https://perceptualui.org/research/datasets/MPIIGaze/). We have been achieving great results on predicting the normalized gaze during inference using our webcam, and we are interested in transforming the normalized gaze to the camera coordinate space so that we can plot the user's point of gaze on the screen.

After reading the README of this repository, I believe that the papers published before [2] do indeed use the scaling factor in the normalized gaze, while papers published after do not use the scaling factor in gaze normalization. So, to my knowledge, the papers that use the MPIIGaze dataset ([1] and [3]) use the scaling factor in the gaze normalization. However, I am wondering if the MPIIGaze dataset available on the website has been updated so that the scaling factor is not applied to the normalized gaze in the "Normalized" folder, as it seems that the MPIIGaze dataset was last updated on May 31, 2018.

Just to clarify, the normalized data in MPIIGaze dataset available from [https://perceptualui.org/research/datasets/MPIIGaze/](https://perceptualui.org/research/datasets/MPIIGaze/) uses the scaling factor in the normalized gaze, correct?

Thank you in advance for all of your help,
Spencer

[1] MPIIGaze: Real-World Dataset and Deep Appearance-Based Gaze Estimation. Zhang et al. TPAMI 2017.
[2] Revisiting Data Normalization for Appearance-Based Gaze Estimation. Zhang et al. ETRA 2018.
[3] Appearance-Based Gaze Estimation in the Wild. Zhang et al. CVPR 2015.

**Comment by xucong-zhang** [comment]
Dear Spencer,

I am glad that our work is helpful to you.

I confirm the MPIIGaze dataset from https://perceptualui.org uses the scaling factors in the normalized gaze. We intended to do that because both [1] and [3] were already published. I forgot why the data was last updated on May 31, 2018, but it should not affect the scaling factors issue.

I am sorry for the confusion it may cause to you. We should have realized this issue earlier.

Best regards,
Xucong

**Comment by 1cookspe** [comment]
Dear Xucong,

Thank you so much for your response and the clarification; it is greatly appreciated!

In order to perform real-time inference on eye images that we acquire from our webcam, we are interested in transforming the predicted gaze vectors from the normalized coordinate space to the camera coordinate space so that we can plot the user's gaze on the screen.
As outlined in [1], we multiply the predicted gaze vector in the normalized coordinate space, `g_n`, by the inverse of `M`, where `M = np.matmul(S, R)`, to acquire the predicted gaze vector in the camera coordinate space, `g_c`, like so:
`g_c = np.matmul(np.linalg.inv(M), g_n)` 

Through our initial experiments, we are calculating the point of intersection on the screen using the normalized gaze vector, `g_n`, by extending the normalized vector from the user's eye to the screen. This has been giving us very good results, as the gaze location that we plot on screen is very close to the user's actual gaze point of intersection. However, we have recently begun to calculate the gaze vector in the camera coordinates, `g_c`, and we are not achieving as accurate results when plotting our gaze on the screen using `g_c`.

Are we required to apply any other transformations to the normalized gaze before finding the intersection between the gaze vector and the screen? We are unsure why extending the normalized gaze to the screen is giving us better results than extending the gaze vector in the camera coordinates to the screen. From Figure 7 in [2], we see that the error in the original data normalization (using the scaling factor) is greater than the error for no data normalization at distances less than 400 mm, but we are conducting our experiments at 600 mm and greater. 

We are wondering if we are calculating or misinterpreting `g_c` incorrectly, causing us to achieve less accuracy than using `g_n` in our point of intersection on the screen. Thank you so much for all of your help on this matter!

Best regards,
Spencer

**Comment by xucong-zhang** [comment]
Dear Spencer,

Your calculation of converting the gaze direction to the camera coordinate system is correct. I don't know why you got worse results after converting it back to the camera coordinate system. My guess is the normalization with the scaling factor makes the gaze direction more centralized, it is maybe why you got better results with it: the gaze direction would not spread.

Best regards,
Xucong
